app/lib/actions/katello/repository/destroy.rb

```diff
@@ -18,7 +18,16 @@ def plan(repository, options = {})
           end
 
           plan_action(ContentViewPuppetModule::Destroy, repository) if repository.puppet?
-          plan_action(Pulp::Repository::Destroy, repository_id: repository.id)
+
+          if SmartProxy.pulp_master.backend_service_type(repository) ==
+            Actions::Pulp3::Abstract::BACKEND_SERVICE_TYPE
+            plan_action(Pulp3::Orchestration::Repository::Delete,
+              repository.id, SmartProxy.pulp_master)
+          else
+            plan_action(Pulp::Repository::Destroy,
+              repository_id: repository.id)
+          end
```

Comment:
Any reason to not use the pulp selector here?  (we might need to adjust some of the actions to have the same signatures for planning).

Reply:
(or possibly wrap it in an a pulp2 orchestration action, even if it just calls Pulp::Repository::Destroy, in order to have a similar signature, and thus usable by pulpselector)

Reply:
@jlsherrill There is a signature difference between Pulp2 and Pulp3 destroy - one takes a repository model and the other takes a repository id, etc. I tried to refactor the pulp 2 action but quickly realized that wasn't a good approach. I'll try out a Pulp2 orchestration action that should work with the PulpSelector.